File: phoenix/phoenix-5.x/src/main/python/TestPhoenix.py
import phoenixdb
import phoenixdb.cursor

# 使用python需在环境中安装phoenix-python驱动，参见https://python-phoenixdb.readthedocs.io/en/latest/
if __name__ == "__main__":
    # url修改为运行环境的QueryServer连接地址
    database_url = 'http://localhost:8765'
    # python客户端仅支持自动提交方式
    conn = phoenixdb.connect(database_url, autocommit=True)

    cursor = conn.cursor()
    cursor.execute("DROP TABLE IF EXISTS users")
    # 创建测试表
    cursor.execute("CREATE TABLE users (id INTEGER PRIMARY KEY, username VARCHAR)")
    # 创建索引
    cursor.execute("CREATE INDEX users_index ON users (username)")

    # 插入一条数据
    cursor.execute("UPSERT INTO users VALUES (?, ?)", (11, 'admin'))
    # 批量插入数据
    param = ((1, "tom"), (2, "alice"), (3, "bob"))
    cursor.executemany("UPSERT INTO users VALUES (?, ?)", param)

    # 查询数据
    cursor.execute("SELECT * FROM users")
    print cursor.fetchall()
    cursor = conn.cursor(cursor_factory=phoenixdb.cursor.DictCursor)
    # query data use index
    cursor.execute("SELECT * FROM users WHERE username='admin'")
    print cursor.fetchone()['ID']

File: phoenix/phoenix-5.x/src/main/go/TestPhoenix.go
package main

import "database/sql"
import "fmt"
import _ "github.com/apache/calcite-avatica-go/v3"

// 运行程序前需要安装phoenix-go驱动，详见https://github.com/apache/calcite-avatica-go
func main() {
  // localhost修改为运行环境的QueryServer主机名或IP
  db, err := sql.Open("avatica", "http://localhost:8765")
  checkErr(err)
  db.Exec("drop table if exist go_test")
  db.Exec("create table go_test(id integer primary key, name varchar)")
  db.Exec("create index go_test_idx on go_test(name)")

  // 使用Prepare批量写入数据
  stm,_ := db.Prepare("upsert into go_test values(?, ?)")
  stm.Exec(111, "admin")
  stm.Exec(112, "root")

  rows, err := db.Query("select * from go_test")
  checkErr(err)
  for rows.Next() {
        var id int
        var name string
        err := rows.Scan(&id, &name)
        checkErr(err)
        fmt.Printf("id : %d\n", id)
        fmt.Printf("username : %s\n", name)
  }
  db.Close()
}

func checkErr(err error) {
  if err != nil {
        panic(err)
  }
}